### PR TITLE
Add TDD framework, description and usage instructions

### LINKED ISSUE #2: Add some tests
I jumped the gun with #1. First we should create some tests that we would like our example code to satisfy, following a test-driven development (TDD) approach.

### LINKED ISSUE #3: Add a description of the example code
When example code is introduced in #1 and #2, we should add some text to the README giving an overview of its functionality.

### LINKED ISSUE #4: Add usage instructions
The README should provide clear instructions on how to use this repository. These instructions should include:
* How to set up the Python environment.
* How to install dependencies.
* How to run the tests.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #2.
Closes #3.
Closes #4.

This PR introduces tests for an as-yet unimplemented function `equation_of_line`, following a test-driven development (TDD) approach.

It also provides an overview on the function and its tests, as well as usage instructions for the repo.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+numpy
```
Comment (TomMelt) [inline requirements.txt:1]: might be worth investigating that it works for `numpy` version 2. Maybe freeze version if not.

**Review by TomMelt — APPROVED** [review]
Looks good to me. Just one comment which is optional.

File: requirements.txt (hunk 1 above)
Comment (jwallwork23) [reply requirements.txt:1]: I just upgraded to latest NumPy and can confirm that everything still works.

**Comment by jwallwork23** [comment]
Thanks @TomMelt!

**Pull request merged**